### PR TITLE
e2e_node: mount_rro: skip if feature gate is disabled; fix error string comparison

### LINKED ISSUE #123848: Failure cluster [cc24ffcf...] [NodeFeature:RecursiveReadOnlyMounts] Mount recursive read-only when the runtime does not support recursive read-only mounts should reject recursive read-only mounts
image### Failure cluster [cc24ffcff22006bc591f](https://go.k8s.io/triage#cc24ffcff22006bc591f)

##### Error text:
```
[FAILED] Timed out after 120.000s.
The matcher passed to Eventually returned the following error:
    <*errors.errorString | 0x40011eb5c0>: 
    pod was expected to be pending, but it is in the state: Succeeded
    {
        s: "pod was expected to be pending, but it is in the state: Succeeded",
    }
In [It] at: k8s.io/kubernetes/test/e2e_node/mount_rro_linux_test.go:218 @ 03/09/24 22:17:24.695

```
#### Recent failures:
[3/10/2024, 6:06:45 PM ci-kubernetes-node-swap-ubuntu-serial](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-swap-ubuntu-serial/1766948619253977088)
[3/10/2024, 5:07:57 PM ci-cgroupv1-containerd-node-arm64-e2e-serial-ec2-eks](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-cgroupv1-containerd-node-arm64-e2e-serial-ec2-eks/1766933771149381632)
[3/10/2024, 4:33:36 PM ci-kubernetes-node-swap-fedora-serial](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-swap-fedora-serial/1766924963056455680)
[3/10/2024, 3:53:42 PM ci-kubernetes-node-arm64-ubuntu-serial](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-node-arm64-ubuntu-serial/1766915148313792512)
[3/10/2024, 2:56:56 PM ci-cgroupv2-containerd-node-arm64-al2023-e2e-serial-ec2-eks](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-cgroupv2-containerd-node-arm64-al2023-e2e-serial-ec2-eks/1766900803433402368)

/kind failing-test

/sig node

### PR DESCRIPTION
**Pull request by AkihiroSuda**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:
Fixes:
- #123848

#### Which issue(s) this PR fixes:

Fixes #123848

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/mount_rro_linux_test.go [hunk 1]
```diff
@@ -135,6 +136,7 @@ var _ = SIGDescribe("Mount recursive read-only [LinuxOnly]", framework.WithSeria
 		}) // Context
 		ginkgo.Context("when the runtime does not support recursive read-only mounts", func() {
 			f.It("should accept non-recursive read-only mounts", func(ctx context.Context) {
+				e2eskipper.SkipUnlessFeatureGateEnabled(features.RecursiveReadOnlyMounts)
```
Comment (AkihiroSuda) [inline test/e2e_node/mount_rro_linux_test.go:139]: (Looks like we can't use this. The test is now always skipped, even with `TEST_ARGS='--service-feature-gates=RecursiveReadOnlyMounts=true --kubelet-flags="--feature-gates=RecursiveReadOnlyMounts=true"`)

File: test/e2e_node/mount_rro_linux_test.go [hunk 2]
```diff
@@ -140,6 +140,9 @@ var _ = SIGDescribe("Mount recursive read-only [LinuxOnly]", framework.WithSeria
 					if supportsRRO(ctx, f) {
 						e2eskipper.Skipf("runtime supports recursive read-only mounts")
 					}
+					if !rroGateIsEnabeld(ctx, f) {
```
Comment (dims) [inline test/e2e_node/mount_rro_linux_test.go:143]: typo alert! - `rroGateIsEnabeld` -> `rroGateIsEnabled`

Comment (AkihiroSuda) [reply test/e2e_node/mount_rro_linux_test.go]: fixed

File: test/e2e_node/mount_rro_linux_test.go (hunk 1 above)
Comment (carlory) [reply test/e2e_node/mount_rro_linux_test.go]: How about just adding one skipper to the SIGDescribe container? for example,

```go
ginkgo.BeforeEach(func(ctx context.Context) {
    e2eskipper.Skipf("feature gate RecursiveReadOnlyMounts is not enabled")
})
```

BTW, which job will run these tests?

Comment (AkihiroSuda) [reply test/e2e_node/mount_rro_linux_test.go:135]: Sorry, my comment was wrong, please ignore

File: test/e2e_node/mount_rro_linux_test.go [hunk 3]
```diff
@@ -123,13 +123,9 @@ var _ = SIGDescribe("Mount recursive read-only [LinuxOnly]", framework.WithSeria
...
-					pod = e2epod.NewPodClient(f).Create(ctx, pod)
-					framework.ExpectNoError(e2epod.WaitForPodContainerToFail(ctx, f.ClientSet, pod.Namespace, pod.Name, 0, "CreateContainerConfigError", framework.PodStartShortTimeout))
-					var err error
-					pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(ctx, pod.Name, metav1.GetOptions{})
-					framework.ExpectNoError(err)
-					gomega.Expect(pod.Status.ContainerStatuses[0].State.Waiting.Message).To(
-						gomega.ContainSubstring("failed to resolve recursive read-only mode: volume \"mnt\" requested recursive read-only mode, but it is not read-only"))
+					_, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+					gomega.Expect(err.Error()).To(
+						gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true"))
```
Comment (AkihiroSuda) [inline test/e2e_node/mount_rro_linux_test.go:128]: (This fix is distinct from #123848)

Comment (carlory) [reply test/e2e_node/mount_rro_linux_test.go:128]: How about this?
```golang
gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true")))
```

or 

```golang
gomega.Expect(err).To(gomega.MatchError(apierrors.IsForbidden, fmt.Sprintf("the pod %s is forbidden: may only be specified when readOnly is true", pod.Name)))
```

Comment (carlory) [reply test/e2e_node/mount_rro_linux_test.go:128]: or 
```golang
gomega.Expect(err).To(gomega.MatchError("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true"))
```

Comment (AkihiroSuda) [reply test/e2e_node/mount_rro_linux_test.go]: Updated to use `gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true")))`

**Pull request merged**